### PR TITLE
Serafina/execution reduction

### LINKED ISSUE #1: Reduce number of executions of QC metric calculation
Currently, the QC metrics are re-calculated whenever the metadata is changed, which is quite often and might lead to problems with large datasets. The metadata calculation should be manually triggered whenever the users starts to perform QC.

A rough draft about how this can be achieved:
1. Do not use `calculate_qc_metrics` in `app.py`. Instead have a `reactive.Value[bool]` which is set to true whenever the `_metadata` or `_adata` is updated.
2. Create a new [`input_task_button`](https://shiny.posit.co/py/api/core/ui.input_task_button.html) in `sliders.py`, which only appears if the mentioned boolean is `true`. When clicked, it performs the `calculate_qc_metrics` as a long-running tasks. On completion, update the `_adata_meta` with the result.

### PR DESCRIPTION
**Pull request by serareif**

### CODE REVIEW

File: src/sliders.py [hunk 1]
```diff
@@ -1,124 +1,153 @@
...
+from helpers import calculate_qc_metrics
+
+#_calculate_metrics_bool = reactive.value(False)
+    
+@module.ui
+def slider_ui():
+    return ui.div(
+        ui.output_ui("slider_sample"),
+        ui.output_ui("slider_filters"),
+        button()
```
Comment (nictru) [inline src/sliders.py]: `ui.output_ui("button")`

File: src/sliders.py [hunk 2]
```diff
@@ -1,124 +1,153 @@
...
+    @render.ui
+    def slider_sample():
+        adata = _adata.get()
+        if adata is None:
+            return
+
+        n_obs = adata.n_obs
+        return ui.input_slider('random_sample_size', 'Random sample size', min(100, n_obs), n_obs, min(10000, n_obs), post=" cells")
+    
+    @render.ui
```
Comment (nictru) [inline src/sliders.py]: Add `@output` decorator

**Review by nictru — CHANGES_REQUESTED** [review]

File: src/sliders.py (hunk 1 above)
Comment (serareif) [reply src/sliders.py]: done

File: src/sliders.py (hunk 2 above)
Comment (serareif) [reply src/sliders.py]: done

**Review by nictru — CHANGES_REQUESTED** [review]
This PR is hard to review since basically everything changed. When looking at the files with `cat app.py -v`, one can see that there are Windows line endings everywhere (indicated by `^M`). You can remove these line using [dos2unix](https://linux.die.net/man/1/dos2unix).

**Comment by nictru** [comment]
I think the way it is implemented now would still block the UI during execution. Check out the explanation [here](https://shiny.posit.co/py/docs/nonblocking.html#true-non-blocking-behavior-with-extendedtask) and make use of the `reactive.extended_task` decorator

**Review by nictru — APPROVED** [review]

**Pull request merged**